### PR TITLE
Automatically focus inputs, or the primary button, in modals.

### PR DESCRIPTION
**Pull request by psybers**
Working so that modals will automatically focus the first input, or for confirmation-style modals, autofocus the "primary" button so you can just hit ENTER when it pops up (or ESC).

There are two React Aria button refactoring PRs remaining.  It looks like #2915 has no overlap.  This PR should be merged after #3163 as it conflicts on one file:

- packages/desktop-client/src/components/mobile/transactions/TransactionEdit.jsx

### CODE REVIEW

**Comment by MatissJanis** [comment]
This is cool, but @joel-jeremy is currently refactoring the `Button` component to `react-aria`.  So perhaps lets hold off on this until his migration is complete?

**Comment by psybers** [comment]
@joel-jeremy Do you have all of the react button refactoring PRs in now?

**Comment by matt-fidd** [comment]
@psybers I believe the relevant react aria button PRs are in now, there's one tiny conflict left.

**Comment by psybers** [comment]
@matt-fidd Conflicts resolved.

File: packages/desktop-client/src/components/modals/ConfirmTransactionDelete.tsx
Comment (matt-fidd) [inline packages/desktop-client/src/components/modals/ConfirmTransactionDelete.tsx:52]: @psybers - I'm not questioning you on this, just a learning moment for me as I get to better grips with React. Why  is this one wrapped in <InitialFocus> instead of an autofocus prop?

Comment (psybers) [reply packages/desktop-client/src/components/modals/ConfirmTransactionDelete.tsx:52]: This uses the `focus()` call on the element, after a short delay.  Which is needed in some cases.

**Review by matt-fidd — APPROVED** [review]

**Comment by matt-fidd** [comment]
Will leave this for a day for any other opinions before I merge it.

**Pull request merged**
